runtime/src/prioritization_fee_cache.rs

```diff
@@ -0,0 +1,788 @@
+            .accumulate_successful_transaction_update_count(successful_transaction_update_count);
+    }
+
+    /// Finalize prioritization fee when it's bank is completely replayed from blockstore,
+    /// by pruning irrelevant accounts to save space, and marking its availability for queries.
+    pub fn finalize_priority_fee(&self, slot: Slot) {
+        self.sender
+            .send(FinalizingServiceUpdate::BankFrozen { slot })
+            .unwrap_or_else(|err| {
+                warn!(
+                    "prioritization fee cache signalling bank frozen failed: {:?}",
+                    err
+                )
+            });
+    }
+
+    /// Internal function is invoked by worker thread to update slot's minimum prioritization fee,
+    /// Cache lock contends here.
+    fn update_transaction(
+        cache: Arc<RwLock<LruCache<Slot, Arc<Mutex<PrioritizationFee>>>>>,
+        slot: &Slot,
+        transaction_fee: u64,
+        writable_accounts: Arc<Vec<Pubkey>>,
+        metrics: Arc<PrioritizationFeeCacheMetrics>,
+    ) {
+        let (block_prioritization_fee, cache_lock_time) =
+            measure!(Self::get_prioritization_fee(cache, slot), "cache_lock_time");
+
+        let (mut block_prioritization_fee, entry_lock_time) =
+            measure!(block_prioritization_fee.lock().unwrap(), "entry_lock_time");
+
+        let (_, entry_update_time) = measure!(
+            block_prioritization_fee.update(transaction_fee, &writable_accounts),
+            "entry_update_time"
+        );
+        metrics.accumulate_total_cache_lock_elapsed_us(cache_lock_time.as_us());
+        metrics.accumulate_total_entry_lock_elapsed_us(entry_lock_time.as_us());
+        metrics.accumulate_total_entry_update_elapsed_us(entry_update_time.as_us());
+    }
+
+    fn finalize_slot(
+        cache: Arc<RwLock<LruCache<Slot, Arc<Mutex<PrioritizationFee>>>>>,
+        slot: &Slot,
+        metrics: Arc<PrioritizationFeeCacheMetrics>,
+    ) {
+        let (block_prioritization_fee, cache_lock_time) =
+            measure!(Self::get_prioritization_fee(cache, slot), "cache_lock_time");
+
+        let (mut block_prioritization_fee, entry_lock_time) =
+            measure!(block_prioritization_fee.lock().unwrap(), "entry_lock_time");
+
+        // prune cache by evicting write account entry from prioritization fee if its fee is less
+        // or equal to block's minimum transaction fee, because they are irrelevant in calculating
+        // block minimum fee.
+        let (_, slot_finalize_time) = measure!(
+            block_prioritization_fee.mark_block_completed(),
+            "slot_finalize_time"
+        );
+        block_prioritization_fee.report_metrics(*slot);
+        metrics.accumulate_total_cache_lock_elapsed_us(cache_lock_time.as_us());
+        metrics.accumulate_total_entry_lock_elapsed_us(entry_lock_time.as_us());
+        metrics.accumulate_total_block_finalize_elapsed_us(slot_finalize_time.as_us());
+    }
+
+    fn finalizing_loop(
```

Comment:
This does more than finalize now. Recommend renaming, perhaps something similar to however you rename the FinalizingServiceUpdate enum

Reply:
agree, changed to `service_thread`